search/binary_search_tree.py

```diff
@@ -5,10 +5,40 @@ def __init__(self, value):
     self.right = None
 
   def depth_first_for_each(self, cb):
-    pass    
+    # get a reference to right and left nodes
+    left = self.left
+    right = self.right
+    #return cb with a value
+    cb(self.value) 
+
+    # if left exists, recur until we hit the bottom
+    if left is not None:
```

Comment:
remember truthy and falsy values from javascript? you could simply write:
`if left:` and `if right:`. they'll give the same result as `if left is not None:`
https://stackoverflow.com/questions/35642809/understanding-javascript-truthy-and-falsy